### PR TITLE
Run rustfmt on libcore/sync folder

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

**Comment by srinivasreddy** [comment]
@nrc  This looks good to me. lets merge it.  r?@nrc

File: src/libcore/sync/atomic.rs [hunk 1]
```diff
@@ -277,7 +277,9 @@ impl AtomicBool {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn store(&self, val: bool, order: Ordering) {
-        unsafe { atomic_store(self.v.get(), val as u8, order); }
+        unsafe {
+            atomic_store(self.v.get(), val as u8, order);
+        }
```
Comment (mattico) [inline src/libcore/sync/atomic.rs]: Hm, why does rustfmt put this on multiple lines...

File: src/libcore/sync/atomic.rs [hunk 2]
```diff
@@ -632,9 +638,7 @@ impl<T> AtomicPtr<T> {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn load(&self, order: Ordering) -> *mut T {
-        unsafe {
-            atomic_load(self.p.get() as *mut usize, order) as *mut T
-        }
+        unsafe { atomic_load(self.p.get() as *mut usize, order) as *mut T }
```
Comment (mattico) [inline src/libcore/sync/atomic.rs]: ...but this on a single line?

Comment (Stebalien) [inline src/libcore/sync/atomic.rs]: Semicolon versus no semicolon? rustfmt appears to avoid `; }`.

Comment (nrc) [inline src/libcore/sync/atomic.rs]: Yeah, Rustfmt puts blocks which are expressions on one line, but not blocks of statements (even if just one)

**Pull request merged**